packages/babel-parser/src/parser/statement.ts

```diff
@@ -351,7 +351,10 @@ export default abstract class StatementParser extends ExpressionParser {
     if (type === tt._of && !containsEsc) {
       // `using of` must start a for-lhs-of statement
       return false;
-    } else if (tokenIsIdentifier(type) && !this.hasFollowingLineBreak()) {
+    } else if (
+      (tokenIsIdentifier(type) || type === tt._void) &&
```

Comment:
Probably a pointless micro-optimization, feel free to ignore. In `tt`s we currently have identifiers right after keywords. We could move `void` to the end of the keywords, so that we can have a single range covering `void` and identifiers.

Reply:
Good observation. There are a few other candidates that can benefit from this micro-optimization, such as `this` used by `tsSkipParameterStart` or `class` used by the `abstract` arm of the `tsParseDeclaration`, but we can at most pick one token unless there is demand for both `isAorBorIdentifier` and `isBorIdentifier`. 

Since discard-binding is still an early proposal and we might bikeshed the syntax type, I tend to leave it as-is. If it reaches stage 4 as-is and becomes very popular, we will optimize it then.